### DISCUSSION TITLE
Wrong Time And Date

### DISCUSSION THREAD

**Discussion by R101G** [rern/rAudio#472] [Fixed Issues]
@rern 
The time and date are not being set correctly on my installation. I think this was so before the current update, and it makes reading the logs a bit confusing!

Also, the wifi regdomain setting does not stick. I do not think it ever has.

rAudio 1 • 20210813
5.10.52-7-ARCH aarch64
Raspberry Pi 4 Model B Rev 1.2
BCM2711 • 3.7GiB

**Comment by R101G** [comment]
Don't know if it helps, rern, but it looks as if NTP stopped working on July 17.  Entering _systemctl restart systemd-timesyncd_ seems to get it going again.

**Comment by R101G** [comment]
Just to update, I am now running a new 64 bit install on a Pi 4. It has been running for 1day 5hours 48 minutes, and the time is now out by 3m 20s. It does not seem to be getting updated.

I was mistaken, `systemctl restart systemd-timesyncd` does not change anything.

**Comment by R101G** [comment]
After a reboot, the system is showing the time and date from when it was installed.

It looks as if timesyncd is not seeing the network. Could this be because I am using a wifi dongle rather than the on-board ethernet or wifi?

**Comment by R101G** [comment]
Looks like this is the same problem https://archlinuxarm.org/forum/viewtopic.php?f=57&t=15463

**Comment by R101G** [comment]
After creating a file _/etc/systemd/network/wireless.network_ containing:

[Match]
Name=wl*

[Network]
DHCP=no

I have rebooted and powered off several times and the time is still in sync. I am not sure if this is coincidence or a fix.

EDIT: Changed DHCP setting to "no", because it competes with netctl settings. It looks as if timesyncd works when using networkd as an alternative to netctl, not with both. Perhaps ntp might be a better choice for rAudio and netctl.

@rern  ?

**Reply by rern** [reply]
`systemctl status systemd-timesyncd` ?

**Reply by R101G** [reply]
rAudio:~  #  systemctl status systemd-timesyncd
* systemd-timesyncd.service - Network Time Synchronization
     Loaded: loaded (/usr/lib/systemd/system/systemd-timesyncd.service; enabled; vendor preset: enabled)
     Active: active (running) since Mon 2021-08-23 10:24:11 BST; 4h 23min ago
       Docs: man:systemd-timesyncd.service(8)
   Main PID: 263 (systemd-timesyn)
     Status: "Initial synchronization to time server 217.114.59.66:123 (pool.ntp.org)."
      Tasks: 2 (limit: 4518)
        CPU: 175ms
     CGroup: /system.slice/systemd-timesyncd.service
             `-263 /usr/lib/systemd/systemd-timesyncd

Aug 08 19:55:45 rAudio systemd[1]: Starting Network Time Synchronization...
Aug 08 19:55:45 rAudio systemd-timesyncd[263]: System clock time unset or jumped backwards, restoring from recorded timestamp: Mon 2021-08-23 10:24:1>
Aug 23 10:24:11 rAudio systemd[1]: Started Network Time Synchronization.
Aug 23 10:24:18 rAudio systemd-timesyncd[263]: Network configuration changed, trying to establish connection.
Aug 23 10:24:36 rAudio systemd-timesyncd[263]: Initial synchronization to time server 217.114.59.66:123 (pool.ntp.org).

**Comment by R101G** [comment]
Just to confirm that I have had no problems since creating a _/etc/systemd/network/wl.network_ file containing just

[Match]
Name=wl*

**Comment by ceejayemm** [comment]
RPI3b+ RAudio i20210911 (32 bit and 64 bit) + wireless connection

The suggested fix above does not work for me. I created the file exactly as described above. Ran systemctl status systemd-timesyncd and date/time was reset as expected. Following a reboot however the wireless network was disconnected and needed to be reconnected via the Networks screen.  Date/Time has not been sync'ed since this change on Sept 15th

I have had to remove the /etc/systemd/network/wl.network file in order to get the network to be reconnected on reboot.

Chris

**Comment by R101G** [comment]
@ceejayemm  I tried quite a few commands before the above file got it working, and something I did might have been significant. Run `systemctl is-enabled systemd-timesyncd` to check it is enabled at boot. I don't see how it would work for Ethernet connections if it wasn't, though, and it doesn't explain why netctl fails.

I am using a fixed IP with netctl, and the DHCP entry in the systemd/network file conflicted with this. It might be different if you use DHCP. I see that the existing .network files have DHCP=Yes in their network sections, so perhaps it is necessary for wireless too.

The basic problem seems to be that Arch Linux now has networkd as its preferred network manager, and this is causing conflicts. Installing the NTP package seems a sure way to bypass the problem, but at the moment timesyncd is working OK for me.

**Comment by cmh714** [comment]
Interesting issue that I have never run across myself....I am wired but for the time issue I ALWAYS run tzselect and set my tz there and have never had issues with having the correct time.....maybe give that a try.....GL!

**Comment by ceejayemm** [comment]
Thanks for your suggestions below.

I run an RPi3B+ with i20210911 with wireless networking and a fixed IP number provided from the router via DHCP
System Time Zone is set to: Europe - London +01:00
Wireless Country is set to: United Kingdom

I have tried all combinations of files called wl,network and wireless.network (in /etc/systemd/network) with all combination of DHCP=yes or DHCP=no The result of all these tests is that if this file exists in any form then the PI will not automatically start the network. I can however manually start the network from the Network screen and once started RAudio-1 works normally. If this file DOES NOT exist in this location then the PI will boot and automatically connect to the wireless network as expected.

I have run tzselect and set the timezone to TZ='Europe/London'; export TZ in the Rune user account .profile file which does seem to correctly set the time when I login via SSH.  I will check again tomorrow if the time is correctly maintained and set if the system is rebooted.  It does seem to stay set if I manually reboot but then I am checking via SSH (so it would be wouldn't it ?)

NTPD doesn't seem to installed by default.  @rern - if this is related to the differences between netctl and networkd  would it be worth including ntpd in the standard packages for RAudio-1 ?

Chris

**Comment by R101G** [comment]
@ceejayemm 
I wonder if using the /boot/wifi file to re-install wifi settings after creating the . network file is the difference? I am pretty sure I did that at some stage. Other than that, the big difference between our installations is that you are using DHCP, and networkd is taking priority over netctl. Networkd cannot connect on its own without the additional SSID and key entries in the .network file, but it shouldn't need to.

In my case, timesyncd seems to look at all the .network files until it finds one which is up, and then uses it. If there is no match for the wifi interface it fails, which is why a wired connection works out of the box and wifi does not.

**Comment by ceejayemm** [comment]
Just applied 20210924 to my RPi 3B+ (wifi and fixed IP via Router setting). Following reboot, time has jumped backwards (was 22 Sept to now 17 Sept):

With Time Zone set to UTC:
      Time:   17:10:36 • 2021-09-17 UTC
Up Time:   5m since 2021-09-17 • 17:05

With Time Zone set to Europe/London:
      Time:   18:12:31 • 2021-09-17 Europe · London
Up Time:   7m since 2021-09-17 • 18:05

RAConservatory:~  #  systemctl status systemd-timesyncd
* systemd-timesyncd.service - Network Time Synchronization
     Loaded: loaded (/usr/lib/systemd/system/systemd-timesyncd.service; enabled; vendor preset: enabled)
     Active: active (running) since Fri 2021-09-17 18:05:20 BST; 2min 58s ago
       Docs: man:systemd-timesyncd.service(8)
   Main PID: 274 (systemd-timesyn)
     Status: "Daemon is running"
      Tasks: 1 (limit: 1056)
        CPU: 199ms
     CGroup: /system.slice/systemd-timesyncd.service
             `-274 /usr/lib/systemd/systemd-timesyncd

Aug 31 14:11:20 RAConservatory systemd[1]: Starting Network Time Synchronization...
Aug 31 14:11:21 RAConservatory systemd-timesyncd[274]: System clock time unset or jumped backwards, restoring from recorded timesta>
Sep 17 18:05:20 RAConservatory systemd[1]: Started Network Time Synchronization.

Not sure what is happening now.

Chris

**Comment by R101G** [comment]
@rern 
Problem not fixed by 20210924 update. After power-off rAudio does not connect on the first attempt, and time is wrong when it is powered up a second time. In my case, disabling ntpd and recreating a .network file corrects the date and time.

The systemd-timesyncd service was still enabled after the update. Is that correct?

**Comment by ceejayemm** [comment]
Switched to 32bit i20210911 on the basis that RAudio2/3 image might be more suited to it (no real evidence to support this, but worth a try). Updated to 20210924 and saw that the NTP package was actually installed this time around. No issues with wireless network connection as I have given up on /etc/systemd/network/*.network files given the issues I had with them earlier - so no such file present. Rebooted and this time the date setting was correct on the System page:

      Time:  17:50:15 • 2021-09-24 Europe · London
Up Time:  7m since 2021-09-24 • 17:42

In SSH, the date command also returned:

RAConservatory:~  #  date
Fri Sep 24 17:44:07 BST 2021

but systemctl status systemd-timesyncd showed something strange:

RAConservatory:~  #  systemctl status systemd-timesyncd
* systemd-timesyncd.service - Network Time Synchronization
     Loaded: loaded (/usr/lib/systemd/system/systemd-timesyncd.service; enabled; vendor preset: enabled)
     Active: active (running) since Sat 2021-09-11 15:37:47 BST; 1 week 6 days ago
       Docs: man:systemd-timesyncd.service(8)
   Main PID: 294 (systemd-timesyn)
     Status: "Daemon is running"
      Tasks: 1 (limit: 2138)
        CPU: 225ms
     CGroup: /system.slice/systemd-timesyncd.service
             `-294 /usr/lib/systemd/systemd-timesyncd

Aug 31 14:12:46 RAConservatory systemd[1]: Starting Network Time Synchronization...
Aug 31 14:12:46 RAConservatory systemd-timesyncd[294]: System clock time unset or jumped backwards, restoring from recorded timesta>
Sep 11 15:37:47 RAConservatory systemd[1]: Started Network Time Synchronization.
Sep 11 15:37:55 RAConservatory systemd[1]: Stopping Network Time Synchronization...
Sep 11 15:37:55 RAConservatory systemd[1]: systemd-timesyncd.service: Deactivated successfully.
Sep 11 15:37:55 RAConservatory systemd[1]: Stopped Network Time Synchronization.

Please note the sequence I actually ran these commands and then the time it took to cut and paste the results into this message explains the slightly differing times above.

I will reboot again in the morning to see if anything differently happens then.

Chris

(Sorry not an answer, just an update but the forum won't let me do anything else but 'Answer')

**Reply by rern** [reply]
What 20210924 update does on boot:
- Check WiFi connection
- Check Internet connection
- Check wired LAN connection
- If no wired LAN connection, sync date-time
- 
Try:
```sh
if ! ifconfig | grep -q 'inet.*broadcast'; then
	echo No WiFi connection
elif ! : >/dev/tcp/8.8.8.8/53; then
	echo No Internet connection
elif ifconfig | grep -A1 ^eth | grep -q 'inet.*broadcast'; then
	echo Wired LAN connected
else
	echo Sync date-time ...
	ntpdate pool.ntp.org
	echo Current date-time:
	date
fi
```

**Reply by ceejayemm** [reply]
@rern 

Thanks. Since switching to the 32 bit RPi 2/3 build the new time sync has worked for me. I have also run the script above and it updated the time by -0.141962 sec !

 if ! ifconfig | grep -q 'inet.*broadcast'; then
> echo No WiFi connection
> elif ! : >/dev/tcp/8.8.8.8/53; then
> echo No Internet connection
> elif ifconfig | grep -A1 ^eth | grep -q 'inet.*broadcast'; then
> echo Wired LAN connected
> else
> echo Sync date-time ...
> ntpdate pool.ntp.org
> echo Current date-time:
> date
> fi
Sync date-time ...
25 Sep 08:58:05 ntpdate[10987]: adjust time server 176.58.109.199 offset -0.141962 sec
Current date-time:
Sat Sep 25 08:58:05 BST 2021

Thanks

Chris

**Comment by ceejayemm** [comment]
@rern 

Oops! Spoke too soon. After rebooting the Pi (from SSH, if it makes any difference ?) the time was set to:

RAConservatory:~  #  date
Sat Sep 11 15:38:01 BST 2021

Running your script above corrects the time:

RAConservatory:~  #  if ! ifconfig | grep -q 'inet.*broadcast'; then
> echo No WiFi connection
> elif ! : >/dev/tcp/8.8.8.8/53; then
> echo No Internet connection
> elif ifconfig | grep -A1 ^eth | grep -q 'inet.*broadcast'; then
> echo Wired LAN connected
> else
> echo Sync date-time ...
> ntpdate pool.ntp.org
> echo Current date-time:
> date
> fi
Sync date-time ...
25 Sep 09:08:58 ntpdate[1374]: adjust time server 129.250.35.251 offset +0.001890 sec
Current date-time:
Sat Sep 25 09:08:58 BST 2021

A reboot, via the web app, seems to have retained the correct time:

RAConservatory:~  #  date
Sat Sep 25 09:11:23 BST 2021

A further reboot, via SSH, also now seems to have kept the correct time:

RAConservatory:~  #  date
Sat Sep 25 09:13:46 BST 2021

Chris

**Comment by R101G** [comment]
@ceejayemm 
I found that some suggested solutions survived a reboot, but not a power-off, so worth checking that too.

Adding the .network file has proved a rock-solid solution for me, but it seems it does not work for DHCP. One thing which would be interesting to know is whether a wired connection still keeps correct time if the en.network and eth.network files are removed or disabled.

**Comment by rern** [comment]
@R101-Github,

It might be this: [systemd-timesyncd: No network connectivity, ...](https://archlinuxarm.org/forum/viewtopic.php?f=7&t=15574#p68029)

Please try:
- Remove the added file
- Get the UPDATE branch

**Comment by R101G** [comment]
@rern 

This problem has disappeared somewhere along the line. I have been running without the extra file for a while now. Either one of your updates or a kernel update solved it for me.

I have applied the new update anyway, and will report any problems.

Thanks.
